### PR TITLE
Drop flutter submodule

### PR DESCRIPTION
**Pull request by domesticmouse**
Devon, I believe this is a fairly complete migration to the downloaded Flutter SDK. I'm going to land this PR as is so I can test the rest of the build and deploy to Cloud Run workflow.

TBR: @devoncarew

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,18 +24,11 @@ RUN pub get --offline
...
-# Clone the flutter repo and set it to the same commit as the flutter submodule.
-RUN git clone https://github.com/flutter/flutter.git
-RUN cd flutter && git checkout $FLUTTER_COMMIT
-
 # Set the Flutter SDK up for web compilation.
-RUN flutter/bin/flutter doctor
-RUN flutter/bin/flutter config --enable-web
-RUN flutter/bin/flutter precache --web --no-android --no-ios --no-linux \
-  --no-windows --no-macos --no-fuchsia
+RUN dart pub run grinder setup-flutter-sdk
```
Comment (devoncarew) [inline Dockerfile:28]: You _should_ be able to shorten this slightly to just `dart run grinder setup-flutter-sdk` (using the new dart tool).

File: flutter-sdk-version.yaml [hunk 2]
```diff
@@ -11,5 +11,5 @@
 # In order to update the SDK manually, run 'dart tool/update_sdk.dart'.
 
 flutter_sdk:
-  channel: beta
-  # version: 1.25.0-8.1.pre
+  # channel: beta
+  version: 1.25.0-8.3.pre
```
Comment (devoncarew) [inline flutter-sdk-version.yaml:15]: Not for this PR, but going forward, do we want master to track a specific flutter version? A flutter channel? Probably worth thinking about for when we want to do something like deploy dev, beta, and stable sdk versions of dartpad.

File: tool/grind.dart [hunk 3]
```diff
@@ -246,33 +244,30 @@ Future _buildStorageArtifacts(Directory dir) async {
 }
 
 @Task('Delete, re-download, and reinitialize the Flutter submodule.')
-void setupFlutterSubmodule() async {
-  final flutterDir = Directory('flutter');
+void setupFlutterSdk() async {
+  await flutterSdkPath.delete(recursive: true);
```
Comment (devoncarew) [inline tool/grind.dart:248]: I don't think it's necessary to nuke the sdk location each time. The git code I wrote should be able to re-target a sdk checkout to the version / channel we want.

File: flutter-sdk-version.yaml (hunk 2 above)
Comment (domesticmouse) [reply flutter-sdk-version.yaml:15]: I need a commit to push the Cloud Build pipeline to build and deploy to Cloud Run. So, for the time being I'm being explicit with version numbers as a way of making sure I have an explicit build trigger PR.

In the future I'll need a way of having stable, beta, and dev builds, probably on three separate branches?

File: Dockerfile (hunk 1 above)
Comment (domesticmouse) [reply Dockerfile:28]: Fixed in https://github.com/dart-lang/dart-services/pull/645. Turns out I had a chicken and egg problem which took some engineering to fix.

File: tool/grind.dart (hunk 3 above)
Comment (domesticmouse) [reply tool/grind.dart:248]: Also fixed in https://github.com/dart-lang/dart-services/pull/645.

**Pull request merged**